src/placeos-frontend-loader/gitremote.cr

```diff
@@ -0,0 +1,197 @@
+require "placeos-compiler/git"
+
+module PlaceOS::FrontendLoader
+  abstract class PlaceOS::GitRemote
+    private alias Git = PlaceOS::Compiler::Git
+
+    def initialize(@ref : String, @folder_name : String)
+    end
+
+    abstract def commits
+
+    abstract def branches : Hash(String, String)
+
+    abstract def tags : Array(String)
+  end
+
+  class Github < GitRemote
```

Comment:
For a clearer hierarchy, put implementation in a file like this `remote/github.cr`, and the interface in `remote.cr`

```suggestion
abstract class PlaceOS::FrontendLoader::Remote
  class Github < Remote
```

To stave off `class is a module, not a class` errors, add the `require "./remote/*"` at the bottom of `remote.cr`